providers/github/connection/connection.go

```diff
@@ -161,5 +165,73 @@ func newGithubTokenClient(conf *inventory.Config) (*github.Client, error) {
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	return github.NewClient(tc), nil
+	return github.NewClient(newGithubRetryableClient(tc)), nil
+}
+
+func newGithubRetryableClient(httpClient *http.Client) *http.Client {
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 5
+	retryClient.Logger = &zeroLogAdapter{}
+
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	retryClient.HTTPClient = httpClient
+
+	retryClient.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		// Default Retry Policy would not retry on 403 (adding 429 for good measure)
+		if resp.StatusCode == 403 || resp.StatusCode == 429 {
+			// Primary and Seconday rate limit
```

Comment:
As I understood the linked docs, there may be another case for the secondary limit.
The `retry-after` header might be set before we get `x-ratelimit-remaining` with zero.

I don't know why there are two steps for the secondary limit, but I read it we should take the other header into account:
> If the retry-after response header is present, you should not retry your request until after that many seconds has elapsed.

Reply:
I understood it differently tbh, that in addition they send that one for compatibility.
But it might be a good idea to explicitly handle it, I'll update (otherwise it should be caught by Default Policy)